eventio/simtel/objects.py

```diff
@@ -1610,51 +1610,43 @@ def __str__(self):
         )
 
     def parse(self):
-        assert_exact_version(self, supported_version=2)
+        assert_max_version(self, 2)
+        version = self.header.version
         self.seek(0)
         byte_stream = BytesIO(self.read())
 
         event = self.header.id
         shower_num = read_int(byte_stream)
         n_tel = read_int(byte_stream)
         n_pes = read_array(byte_stream, 'i4', n_tel)
-        n_pixels = read_array(byte_stream, 'i4', n_tel)
-
-        # NOTE:
-        # I don't see how we can speed this up easily since the length
-        # of this thing is not known upfront.
-
-        # pixel_pe: a list (running over telescope_id)
-        #         of 2-tuples: (pixel_id, pe)
-        pixel_pe = []
-        mask = (n_pixels > 0) & (n_pes > 0)
 
-        for n_pe, n_pixels in zip(n_pes[mask], n_pixels[mask]):
-            non_empty = read_short(byte_stream)
-            pixel_id = read_array(byte_stream, 'i2', non_empty)
-            pe = read_array(byte_stream, 'i4', non_empty)
-            pixel_pe.append((pixel_id, pe))
-
-        photons = read_array(byte_stream, 'f4', n_tel)
-        photons_atm = read_array(byte_stream, 'f4', n_tel)
-        photons_atm_3_6 = read_array(byte_stream, 'f4', n_tel)
-        photons_atm_qe = read_array(byte_stream, 'f4', n_tel)
-        photons_atm_400 = read_array(byte_stream, 'f4', n_tel)
-
-        return {
+        result = {
             'event': event,
             'shower_num': shower_num,
             'n_tel': n_tel,
             'n_pe': n_pes,
-            'n_pixels': n_pixels,
-            'pixel_pe': pixel_pe,
-            'photons': photons,
-            'photons_atm': photons_atm,
-            'photons_atm_3_6': photons_atm_3_6,
-            'photons_atm_qe': photons_atm_qe,
-            'photons_atm_400': photons_atm_400,
         }
 
+        # this code was never used according to K. Bernlöhr
+        # so we raise an error in case this happens
+        n_pixels = read_array(byte_stream, 'i4', n_tel)
```

Comment:
I think in this comment the phrases "this code" and "this happens" could be improved slightly. 
I'd propose something like this:
```
# n_pixels should never contain non-zero values according to K. Berlöhr (maybe add a date here?)
# so we raise an error in case this happens.
```

Or so.